File: Source/FlashCards/App/FlashCardApplication.ts
enum FactoidType {
    Text,
    Unicode,
    Audio,
    Image,
    Video
}

interface Entity {
    guid: string;
}

interface FactoidCategory extends Entity {
    description: string;
    flashCardApplication: FlashCardApplication;
    flashCardApplicationGuid: string;
}

interface FlashCard extends Entity {
    factoids: Factoid[];
    flashDeck: FlashDeck;
    flashDeckGuid: string;
}

interface FlashDeck extends Entity {
    flashCards: FlashCard[];
    factoidCategories: FactoidCategory[];
    flashCardApplication: FlashCardApplication;
    flashCardApplicationGuid: string;
}

interface Factoid extends Entity {
    factoidType: FactoidType;
    factoidCategory: FactoidCategory;
    flashCard: FlashCard;
    flashCardGuid: string;
    fact: string;
}

interface FlashCardPlayerOptions extends Entity {
    flashDeck: FlashDeck;
    displayCategories: FactoidCategory[];
    evaluationCategory: FactoidCategory;
    flashCardApplication: FlashCardApplication;
    flashCardApplicationGuid: string;
}

interface FlashCardApplication extends Entity {
    name: string;
    flashCardPlayerOptionsList: FlashCardPlayerOptions[];
    flashDecks: FlashDeck[];
    factoidCategories: FactoidCategory[];
}

interface FlashCardSpaState {
    flashCardApplication: FlashCardApplication;

}

File: Source/FlashCards/Entities/FlashCardApplication.cs
namespace Tft.FlashCards.Entities
{
	using System;
	using System.Collections.Generic;
	using System.Linq;
	using System.Web;

	public class FlashCardApplication:Entity
	{
    public FlashCardApplication()
    {
      FlashDecks = new List<FlashDeck>();
      FactoidCategories = new List<FactoidCategory>();
      FlashCardPlayerOptionsList = new List<FlashCardPlayerOptions>();
    }

    public string Name { get; set; }    
    public virtual List<FlashCardPlayerOptions> FlashCardPlayerOptionsList { get; set; }
    public virtual List<FlashDeck> FlashDecks { get; set; }
    public virtual List<FactoidCategory> FactoidCategories { get; set; }
  }
}
